src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1422,10 +1426,137 @@ public override IOperation VisitMethodBodyOperation(IMethodBodyOperation operati
+            {
+                if (param.IsNullChecked)
+                {
+                    var check = GenerateNullCheckForParameter(param, operation.Syntax, ((Operation)operation).OwningSemanticModel);
+                    _currentStatement = check;
+                    VisitConditional(check, captureIdForResult);
+                }
+            }
+            _currentStatement = temp;
+        }
+
+        private ConditionalOperation GenerateNullCheckForParameter(IParameterSymbol parameter, SyntaxNode syntax, SemanticModel semanticModel)
+        {
+            Debug.Assert(parameter.Language == LanguageNames.CSharp);
+            Optional<object> constantValue = default(Optional<object>);
+            var paramReference = new ParameterReferenceOperation(parameter, semanticModel, syntax, parameter.Type, constantValue, isImplicit: true);
+            var boolType = _compilation.GetSpecialType(SpecialType.System_Boolean);
+            IOperation conditionOp;
+            if (ITypeSymbolHelpers.IsNullableType(parameter.Type))
+            {
+                conditionOp = new UnaryOperation(
+                    UnaryOperatorKind.Not,
+                    new InvocationOperation(
+                        targetMethod: ((IMethodSymbol)_compilation.CommonGetSpecialTypeMember(SpecialMember.System_Nullable_T_get_HasValue)).Construct(parameter.Type),
+                        instance: paramReference,
+                        isVirtual: false,
+                        arguments: ImmutableArray<IArgumentOperation>.Empty,
+                        semanticModel,
+                        syntax,
+                        boolType,
+                        constantValue,
+                        isImplicit: true),
+                    isLifted: false,
+                    isChecked: false,
+                    operatorMethod: null,
+                    semanticModel,
+                    syntax,
+                    boolType,
+                    constantValue,
+                    isImplicit: true);
+            }
+            else
+            {
+                conditionOp = new BinaryOperation(
+                    BinaryOperatorKind.Equals,
+                    paramReference,
+                    new LiteralOperation(semanticModel, syntax, parameter.Type, ConstantValue.Null, isImplicit: true),
+                    isLifted: false,
+                    isChecked: false,
+                    isCompareText: false,
+                    operatorMethod: null,
+                    unaryOperatorMethod: null,
+                    semanticModel,
+                    syntax,
+                    boolType,
+                    constantValue,
+                    isImplicit: true);
+            }
+            var argumentNullExceptionMethodSymbol = (IMethodSymbol)_compilation.CommonGetWellKnownTypeMember(WellKnownMember.System_ArgumentNullException__ctorString);
+            var argumentNullExceptionType = argumentNullExceptionMethodSymbol?.ReturnType;
```

Comment:
>argumentNullExceptionMethodSymbol?.ReturnType [](start = 44, length = 45)

The .ctor return type is `void`. This should be `argumentNullExceptionMethod.ContainingType`.